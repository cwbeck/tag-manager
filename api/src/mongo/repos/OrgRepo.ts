import Repo from '../abstractions/Repo';
import { injectable } from 'inversify';
import Org from '../models/Org';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class OrgRepo extends Repo<Org> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                _name: 1,
            },
            unique: true,
        },
    ];
}
