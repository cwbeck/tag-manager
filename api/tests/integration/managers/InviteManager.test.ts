import ManagerTestUtils from '../../utils/ManagerTestUtils';
import TestUtils from '../../utils/TestUtils';
import OrgRepo from '../../../src/mongo/repos/OrgRepo';
import InviteRepo from '../../../src/mongo/repos/InviteRepo';
import Mock = jest.Mock;

describe('InviteManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetch invite through org', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();
        ManagerTestUtils.mockAsLoggedUser();

        const mockOrg = TestUtils.buildMockOrg();
        const mockOrgRepo = {} as OrgRepo;
        mockOrgRepo.findByIdThrows = jest.fn().mockResolvedValue(mockOrg);
        const mockSutModel = TestUtils.buildMockInvite();
        const mockSutRepo = {} as InviteRepo;
        mockSutRepo.find = jest.fn().mockResolvedValue([mockSutModel]);
        mockSutRepo.findOneThrows = jest.fn().mockResolvedValue(mockSutModel);
        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockImplementation((model: any) => {
            if (model.name === 'Invite') {
                return mockSutRepo;
            }
            return mockOrgRepo;
        });

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getOrg(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    name
                    invites {
                        id
                        email
                        org_permissions {
                            can_view
                            can_create
                            can_edit
                            can_delete
                            is_admin
                        }
                        created_at
                        updated_at
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
