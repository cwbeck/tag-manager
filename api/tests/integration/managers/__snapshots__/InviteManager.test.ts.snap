// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteManager fetch invite through org 1`] = `
Object {
  "data": Object {
    "getOrg": Object {
      "id": "5edf4c48e9615d9a1c66bbbb",
      "invites": Array [
        Object {
          "created_at": 1608175440000,
          "email": "test@mail.com",
          "id": "5edf4c48e9615d9a1c66bddd",
          "org_permissions": Object {
            "can_create": false,
            "can_delete": false,
            "can_edit": false,
            "can_view": true,
            "is_admin": false,
          },
          "updated_at": 1608441840000,
        },
      ],
      "name": "testOrg",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
